### DISCUSSION TITLE
动态库检索

### DISCUSSION THREAD

**Discussion by davidxiaozhi** [xmake-io/xmake#3443] [Q&A]
@waruqi ru
add_requires("torch",{system = true }) 
add_requires("tbb",{system = true })   system =true 会去父 xmake.lua 当中全局设置的 目录中查找么,$(projectdir)/third_part/libtorch/lib 这个目录下是存在动态库 和静态库的
add_linkdirs("$(projectdir)/third_part/kv/lib",
    "$(projectdir)/third_part/libtorch/lib",
    "/App/conda/envs/conda_xmake/lib",
    "/App/conda/envs/conda_xmake/lib64")

问题 1: 我使用的 add_package 这个是不是必须本地存在 package 配置,如果我改用 add_links 是不是解决问题, 使用 add_package 会自动 add_links 么
问题 2: add_requires("tbb",{system = true })  会依据 link_dirs 查找么,  如果不是 依据什么进行查找

**Comment by waruqi** [comment]
> system =true 会去父 xmake.lua 当中全局设置的 目录中查找么

只从系统库路径中找，/usr 或者 pkgconfig 。。。

> 问题 1: 我使用的 add_package 这个是不是必须本地存在 package 配置,如果我改用 add_links 是不是解决问题, 使用 add_package 会自动 add_links 么

不要混用，add_requires 走系统库找，add_package 集成的就是系统库 linkdirs/links 。。用了包，就不要用 add_linkdirs/add_links

> 问题 2: add_requires("tbb",{system = true }) 会依据 link_dirs 查找么, 如果不是 依据什么进行查找

pkgconfig or find /usr/lib
